webapp/package.json

```diff
@@ -3,7 +3,7 @@
   "version": "2.14.4",
   "private": true,
   "engines": {
-    "node": "^14"
+    "node": "18.12.0"
```

Comment:
Je pense qu'il vaut mieux fixer la version à `18.16.1`, qui est la dernière version supportée en LTS.

> D'ailleurs, pourquoi `18.12.0` ?

Reply:
Il convient aussi de mettre à jour la version de Node.js dans le fichier de configuration du GitHub Actions `.github/workflows/main.yml`. Par la même occasion, on peut aussi mettre à jour les actions `actions/checkout` et `actions/setup-node` :

```diff
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-20.04
     steps:
       - name: Checkout github repo (+ download lfs dependencies)
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Set up Python
         uses: actions/setup-python@v1
         with:
@@ -38,7 +38,7 @@ jobs:
 
     steps:
       - name: Checkout github repo (+ download lfs dependencies)
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Set up Python
         uses: actions/setup-python@v1
         with:
@@ -68,11 +68,11 @@ jobs:
         os: [ ubuntu-20.04 ]
     steps:
       - name: Checkout github repo
-        uses: actions/checkout@v1
+        uses: actions/checkout@v3
       - name: Set up Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
-          node-version: 14.x
+          node-version: '18.16.1'
       - name: Install dependencies
         run: npm install
         working-directory: webapp
@@ -100,7 +100,7 @@ jobs:
     runs-on: ubuntu-20.04
     needs: [python-test, npm-test]
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Download python coverage report
       uses: actions/download-artifact@v3
       with:
~
```

Reply:
La documentation doit aussi être mise à jour. Peux-tu corriger le numéro de version de Node.js dans `docs/install/0-INSTALL.md`.

```diff
@@ -7,7 +7,7 @@ The front end is a [React](https://reactjs.org/) web application. A local build
 
 Requirements : 
 - python : 3.8.x
-- node : 14.x
+- node : 18.16.1
 
 1. First clone the projet:
```

Reply:
La 18.12 c'est ma version en local, j'avais mis ça pour éviter le warning, du coup ce sera bien 18.16.1. Pour le reste ça ne concerne pas cette PR je pense qu'il vaut mieux éviter de mélanger, je ferais une branche à part pour tester sur l'intégration puis si tout est ok une PR sur dev pour mettre à jour les versions dans les actions github. @laurent-laporte-pro

Reply:
Je pense que ça fait partie de la PR justement, sinon les tests unitaires ne passent pas.

Habituellement, une User Story doit être entièrement testée et documentée…

Reply:
Tu peux faire un commit du genre "build(GitHub): use Node.js v18.16.1 for CI"